### ISSUE TITLE
spaces as %20

### ISSUE THREAD

**Issue by cvogt** [technosophos/dashing#9]
I don't know why, but in the lower left corner view, where the sections of the currently opened page show up, all spaces are replaced by %20, same in the breadcrumb navigation on top. In other places like the normal tree on the left, the spaces are correct. The SQLite database looks fine as well. I can't tell a difference to the docsets that comes with dash. It's weird

**Comment by technosophos** [comment]
I assume that this just started happening since I added the URL encoding? I wonder if there's some place where they're not being decoded correctly.

**Comment by cvogt** [comment]
Oh, I see. It takes the links from the html, not the database.

So here's the problem: https://github.com/technosophos/dashing/commit/797b810c03ecffc1c4603cc5f57b269593b86d99

it's double escaped, once with url.URL, and then again with QueryEscape. This leads to

"A Modernized Scala REPL" becoming `//apple_ref/cpp/Section/A%2520Modernized%2520Scala%2520REP`, which is one too many escapings " " became "%20" on first encoding and then ""%2520" on second.

Here is a related older commit 
https://github.com/technosophos/dashing/commit/10aad346035bac1666c8582798e8738c258a5ed2

So how to fix it? The older commit indicates that using QueryEscape uses the `+` encoding for spaces, which is valid, but seems like dash or apple do not properly decode it back into spaces (I am guessing here, but the commit message seems to indicate that). However url.URL does not seem to escape `/` at all. Meaning either one or the combo of both are broken for our use case.

The source code (https://golang.org/src/net/url/url.go?s=4483:4516#L181) seems to me like there is no built-in way, but it seems you can just to QueryEscape **only** and then replace `+` with `%20`,

so basically `name = url.QueryEscape(name).replace("+","%20")` or whatever the syntax for replace is in go. Since previously existing `+` characters have already been escaped using %<number>, this seems to be correct.

**Comment by technosophos** [comment]
Alright, let's give it a shot and see if that fixes it.

**Comment by cvogt** [comment]
@technosophos almost :). This should do it: https://github.com/technosophos/dashing/pull/12

**Comment by cvogt** [comment]
confirmed it works now :)! Thanks a lot!
